trade_remedies_api/audit/management/commands/trs_kpis.py

```diff
@@ -0,0 +1,112 @@
+import os
+
+from typing import Any, Optional, List, Union
+
+from django.core.management.base import BaseCommand, CommandParser
+
+from security.constants import SECURITY_GROUPS_PUBLIC
+
+from core.models import UserProfile
+from cases.models import Case, Submission, SubmissionDocument
+
+
+class Command(BaseCommand):
+
+    help = (
+        "Produce report related to key metrics relating to the TRS"
+        "Accounts: Email verified public user accounts"
+        "Sessions: Successful public user log-ins (in period)"
+        "Case registrations requested"
+        "Submissions: Case Submissions"
+        "Files uploaded by public users"
+        "Files downloaded by public users"
+    )
+
+    def add_arguments(self, parser: CommandParser) -> None:
+        # Named (optional) arguments
+
+        parser.add_argument("--date_from", action="store")
+        parser.add_argument("--date_to", action="store")
+        parser.add_argument("--outpath", action="store")
+
+    def handle(self, *args: Any, **options: Any) -> Optional[str]:
+        """
+        _summary_:
+            Command to receive regular reports on key metrics relating to the TRS
+
+            e.g usage:
+                $ python manage.py trs_kpis --date_from 2012-12-12 --date_to 2013-12-12
+
+        _options_:
+            date_to:    ->   YYYY-MM-DD
+            date_from:  ->   YYYY-MM-DD
+            outpath:    ->   e.g. => /tmp/
+        """
+
+        date_range: Union[List[str], None]
+
+        if options["date_from"] and options["date_to"]:
+            date_range = (
+                options["date_from"],
+                options["date_to"],
+            )
+
+            public_users_count = UserProfile.objects.filter(
```

Comment:
Then in the queries below, you can do the same query regardless of whether or not you want the date range. e.g.

```suggestion
       public_users_count = UserProfile.objects.filter(
                email_verified_at__isnull=False,
                user__groups__name__in=SECURITY_GROUPS_PUBLIC,
                **additional_filters
            ).count()
```